text/0000-build-std/4-stage-1a.md

```diff
@@ -0,0 +1,1061 @@
+# Proposal
+[proposal]: #proposal
+
+Cargo configuration will contain a new key `build-std` under the `[build]`
+section ([?][rationale-build-std-in-config]), permitting one of two values -
+"never" ([?][rationale-build-std-never]) or "always", defaulting to "never":
+
+```toml
+[build]
+build-std = "always" # or `never`
+```
+
+`build-std` can also be specified in the `[target.<triple>]` and
+`[target.<cfg>]` sections ([?][rationale-build-std-target-section]):
+
+```toml
+[target.aarch64-unknown-illumos]
+build-std = "always" # or `never`
+```
+
+The `build-std` configuration locations have the following precedence
+([?][rationale-build-std-precedence]):
+
+1. `[target.<triple>]`
+2. `[target.<cfg>]`
+3. `[build]`
```

Comment:
This precedence means that target triple/cfg options in low-precedence config locations can override users explicitly asking for `build.build-std=always` in high precedence locations, like through `--config`, which is unintuitive at best. It's really hard to override those settings in the global config.